integration_tests/offboard_position.cpp

```diff
@@ -0,0 +1,111 @@
+#include <iostream>
+#include <cmath>
+#include "integration_test_helper.h"
+#include "dronecode_sdk.h"
+#include "plugins/action/action.h"
+#include "plugins/telemetry/telemetry.h"
+#include "plugins/offboard/offboard.h"
+#include "plugins/mission/mission.h"
+
+using namespace dronecode_sdk;
+
+TEST_F(SitlTest, OffboardPositionNED)
+{
+    DronecodeSDK dc;
+
+    ConnectionResult ret = dc.add_udp_connection();
+    ASSERT_EQ(ConnectionResult::SUCCESS, ret);
+
+    // Wait for system to connect via heartbeat.
+    std::this_thread::sleep_for(std::chrono::seconds(2));
+
+    ASSERT_TRUE(dc.system().has_autopilot());
+
+    System &system = dc.system();
+    auto telemetry = std::make_shared<Telemetry>(system);
+    auto action = std::make_shared<Action>(system);
+    auto offboard = std::make_shared<Offboard>(system);
+    auto mission = std::make_shared<Mission>(system);
+
+    while (!telemetry->health_all_ok()) {
+        std::cout << "waiting for system to be ready" << std::endl;
+        std::this_thread::sleep_for(std::chrono::seconds(1));
+    }
+
+    Action::Result action_ret = action->arm();
+    ASSERT_EQ(Action::Result::SUCCESS, action_ret);
+
+    action_ret = action->takeoff();
+    ASSERT_EQ(Action::Result::SUCCESS, action_ret);
+
+    std::this_thread::sleep_for(std::chrono::seconds(5));
+
+    // Send it once before starting offboard, otherwise it will be rejected.
+    offboard->set_position_ned({0.0f, 0.0f, -10.0f, 0.0f});
+
+    Offboard::Result offboard_result = offboard->start();
+
+    EXPECT_EQ(offboard_result, Offboard::Result::SUCCESS);
+
+    offboard->set_position_ned({0.0f, 0.0f, -10.0f, 90.0f});
+    std::this_thread::sleep_for(std::chrono::seconds(1));
+
+    {
+        const float radius = 10.0f;
+        const float step = 0.01f;
+        for (float i = 0.0f; i <= 2 * static_cast<float>(M_PI); i += step) {
```

Comment:
Two changes I would suggest here:

1. Rename `i` to `angle` because `i` is usually an integer, so this can be confusing.
2. And I would get rid of the cast, all you need is:
```
2.0f * M_PI_F
```
    or:
```
constexpr float full_circle = 2.0f * M_PI_F;
for (...; angle <= full_circle; ++angle) {
```
    or:
```
static_cast<float>(M_2_PI)
```